life/src/main.rs

```diff
@@ -0,0 +1,475 @@
+
+const ROWS: usize = 3;
+const COLS: usize = 3;
+const OFFSETS: [i8; 3] = [-1, 0, 1];
+const DEAD: bool = false;
+const ALIVE: bool = true;
+const FRAME_DELAY: f64 = 0.3;
+const WINDOW_HEIGHT: i32 = 700;
+const WINDOW_WIDTH: i32 = 800;
+const GRID_HEIGHT: f32 = WINDOW_HEIGHT as f32 - BUTTON_TAB_HEIGHT;
+const GRID_WIDTH: f32 = 800.;
+const BUTTON_TAB_HEIGHT: f32 = 100.;
+const INSTRUCTIONS_TEXT: &str = "To start or stop press 'ENTER'";
+const INSTRUCTIONS_TEXT_X: f32 = 200.;
+const INSTRUCTIONS_TEXT_Y: f32 = 650.;
+const INSTRUCTIONS_TEXT_SIZE: f32 = 30.;
+
+
+/// Verifies if the position represented by the point ('row_i', 'col_i') is inside
+/// the matrix. 
+/// 
+/// # Arguments
+/// 
+/// - `row_i`: Represents the row in the matrix of the point.
+/// - `col_i`: Represents the column in the matrix of the point.
+/// 
+/// # Returns
+/// 
+/// A `bool` indicating if this is true or not.
+fn is_position_valid(row_i: i8, col_i: i8) -> bool {
+    (row_i >= 0 && row_i < ROWS as i8) && (col_i >= 0 && col_i < COLS as i8)
+}
+
+/// Verifies if 2 points on the matrix are the same point. 
+/// 
+/// # Arguments
+/// 
+/// - `row_i_1`: Represents the row in the matrix of point 1.
+/// - `col_i_1`: Represents the column in the matrix of point 1.
+/// - `row_i_2`: Represents the row in the matrix of point 2.
+/// - `col_i_2`: Represents the column in the matrix of point 2.
+/// 
+/// # Returns
+/// 
+/// A `bool` indicating if this is true or not.
+fn is_different_cell(row_i_1: i8, col_i_1: i8, row_i_2: i8, col_i_2: i8) -> bool {
+    row_i_1 != row_i_2 || col_i_1 != col_i_2
+}
+
+/// Changes the state of specific cells in the matrix.
+/// 
+/// # Arguments
+/// 
+/// - `cells`: A Vec which contains tuples that represent the cells in the amtrix to modify.
+/// - `state`: Represents wether the cell is alive or dead.
+/// - `matrix`: An array with arrays that represent the matrix which contains every cell.
+fn manage_cell_state(cells: Vec<(usize, usize)>, state: bool, matrix: &mut [[bool; COLS]; ROWS]) {
+    for (row_i, col_i) in cells {
+        matrix[row_i][col_i] = state;
+    }
+}
+
+/// Checks the eight neighbours of a cell and counts how many of them are alive.
+/// 
+/// # Arguments
+/// 
+/// - `col_i_1`: Represents the column in the matrix of the cell.
+/// - `row_i_1`: Represents the row in the matrix of the cell.
+/// - `matrix`: An array with arrays that represent the matrix which contains every cell.
+/// 
+/// # Returns
+/// 
+/// An `i32` that represents the number of alive neighbours.
+fn check_cell_alive_neighbours(col_i: usize, row_i: usize, matrix: &[[bool; COLS]; ROWS]) -> i32 {
+    let mut alive_neighbours = 0;
+    // This parsing is done here so we don't do it in every loop 
+    let parsed_row_i = row_i as i8;
```

Comment:
Consider whether u want usize or i8 for cells, try to make it more uniform to avoid casts.